File: src/java/3214.java
import java.io.IOException;
import java.util.Scanner;
 
public class Main {
    public static void main(String[] args) throws IOException {
        int e, f, c;
        int posse, comprou, sobrou, tomou;
        Scanner input = new Scanner(System.in);
        
        e = input.nextInt();
        f = input.nextInt();
        c = input.nextInt();
        
        posse = e + f;
        comprou = posse / c;
        sobrou = posse - (comprou * c);
        tomou = comprou;
        
        while (posse >= c) {
            posse = comprou + sobrou;
            comprou = posse / c;
            sobrou = posse - (comprou * c);
            tomou += comprou;
        }
        
        System.out.println(tomou);
    }
}

File: src/py/3214.py
efc = input().split()

e = int(efc[0])
f = int(efc[1])
c = int(efc[2])

posse = e + f
comprou = posse // c
sobrou = posse - (comprou * c)
tomou = comprou

while posse >= c:
    posse = comprou + sobrou
    comprou = posse // c
    sobrou = posse - (comprou * c)
    tomou += comprou

print(tomou)
